### ISSUE TITLE
Configuration file dataset config always uses IID splitter

### ISSUE THREAD

**Issue by MarkHolmstrom** [lishenghui/blades#77]
It is not possible to test non-IID data splits by editing the .yaml configuration files as the IID splitter is always used. 

To replicate, run `blades/tuned_examples/fedavg_cifar10_resnet_noniid.yaml` where 
```
       iid:
        grid_search: [False]
       alpha:
        grid_search: [0.1, 0.5, 1, 100]
```
The resulting trials use the IID data splitter and have the same results for different alpha values.

**Comment by MarkHolmstrom** [comment]
Using the `partitioner_config` configuration as in the new examples works. Addressed in #81

**Comment by lishenghui** [comment]
> Using the `partitioner_config` configuration as in the new examples works. Addressed in #81

Thank you so much @MarkHolmstrom for your contribution! The non-IID partitioner is running smoothly now. I am now closing this issue. I appreciate your reported issue and the modifications you provided. Looking forward to possibly collaborating again in the future!

**Comment by MarkHolmstrom** [comment]
I am looking forward to it as well as I plan to use this library for a research project.
